activerecord/test/cases/finder_test.rb

```diff
@@ -1144,6 +1144,23 @@ def test_find_without_primary_key
       Matey.find(1)
     end
   end
+  
+  def test_first_and_last_with_limit_for_order_without_primary_key
+    # While Topic.first should impose an ordering by primary key,
+    # Topic.limit(n).first should not
+    assert_sql(/^((?!ORDER).)*$/) { Post.limit(1).first }
+    assert_sql(/^((?!ORDER).)*$/) { Post.limit(1).last }
```

Comment:
@sgrif Oh, I was basing it on a similar test `test_find_doesnt_have_implicit_ordering` which I saw in finder_test.rb at https://github.com/rails/rails/blob/master/activerecord/test/cases/finder_test.rb#L307-L309.

Part of the reason I wanted an explicit SQL test like that is that in all databases except Postgres, the default order in which records are returned _is_ ascending by primary key. So the tests based on the results were only failing in Postgres, and only because of how it reacts to the `touch` command (which I would imagine could change in future versions of Postgres). So, I wanted a spec that would fail in all backends, which led me to make the assertion about the SQL itself.

That said, we can remove those SQL assertions and we'll still have a test that fails in current versions of Postgres. If that seems like the better way to go and you'd like me to remove those two SQL tests, let me know and I can add a commit to the PR branch that does that.